File: server/src/main/java/by/brawl/service/AccountServiceImpl.java
package by.brawl.service;

import by.brawl.entity.Account;
import by.brawl.repository.AccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 * Default class description.
 *
 * @author P.Sinitsky.
 *         Created on 25.03.2017.
 */
@Service
class AccountServiceImpl implements AccountService {

    private AccountRepository repository;

    @Autowired
    public AccountServiceImpl(AccountRepository repository) {
        this.repository = repository;
    }

    @Override
    public Account findByEmail(String email) {
        return repository.findByEmail(email);
    }
}

File: server/src/main/java/by/brawl/service/AccountServiceImpl.kt
package by.brawl.service

import by.brawl.repository.AccountRepository
import org.springframework.stereotype.Service

/**
 * Default class description.

 * @author P.Sinitsky.
 * *         Created on 25.03.2017.
 */
@Service
internal class AccountServiceImpl
constructor(private val repository: AccountRepository) : AccountService {

    override fun findByEmail(email: String) = repository.findByEmail(email) ?:
            throw IllegalArgumentException("Account with email $email does not exists!")

}